python/paddle/fluid/tests/unittests/test_setitem_op.py

```diff
@@ -0,0 +1,204 @@
+#   Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Test setitem op in static mode
+
+from __future__ import print_function
+
+import unittest
+import numpy as np
+
+import paddle
+
+
+class TestSetitemBase(unittest.TestCase):
+    def setUp(self):
+        paddle.enable_static()
+        self.set_value()
+        self.set_dtype()
+        self.shape = [2, 3, 4]
+        self.data = np.ones(self.shape).astype(self.dtype)
+        self.program = paddle.static.Program()
+
+    def set_value(self):
+        self.value = 6
+
+    def set_dtype(self):
+        self.dtype = "float32"
+
+    def _call_setitem(self, x):
+        x[0, 0] = self.value
+
+    def _get_answer(self):
+        self.data[0, 0] = self.value
+
+
+class TestSetitemApi(TestSetitemBase):
+    def test_api(self):
+        with paddle.static.program_guard(self.program):
+            x = paddle.ones(shape=self.shape, dtype=self.dtype)
+            self._call_setitem(x)
+
+        exe = paddle.static.Executor(paddle.CPUPlace())
+        out = exe.run(self.program, fetch_list=[x])
+
+        self._get_answer()
+        self.assertTrue(
+            (self.data == out).all(),
+            msg="\nExpected res = \n{}, \n\nbut received : \n{}".format(
+                self.data, out))
+
+
+# 1. Test different type of item: int, python slice
+class TestSetitemItemInt(TestSetitemApi):
+    def _call_setitem(self, x):
+        x[0] = self.value
+
+    def _get_answer(self):
+        self.data[0] = self.value
+
+
+class TestSetitemItemSlice(TestSetitemApi):
+    def _call_setitem(self, x):
+        x[0:2] = self.value
+
+    def _get_answer(self):
+        self.data[0:2] = self.value
+
+
+class TestSetitemItemSlice2(TestSetitemApi):
+    def _call_setitem(self, x):
+        x[0:-1] = self.value
+
+    def _get_answer(self):
+        self.data[0:-1] = self.value
+
+
+class TestSetitemItemSlice3(TestSetitemApi):
+    def _call_setitem(self, x):
+        x[0:-1, 0:2] = self.value
+
+    def _get_answer(self):
+        self.data[0:-1, 0:2] = self.value
+
+
+class TestSetitemItemSlice4(TestSetitemApi):
+    def _call_setitem(self, x):
+        x[0:, 1:2, :] = self.value
+
+    def _get_answer(self):
+        self.data[0:, 1:2, :] = self.value
+
+
+# 2. Test different type of value: int, float, numpy.ndarray, Tensor
```

Comment:
It seems `create_test_value_numpy` only test int numpy, why you commented test different types?

Reply:
Thanks, added unittests for different dtypes. (Here just means type but not dtype. But exactly it is a lack of unittests for different dtype. )